cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt

```diff
@@ -726,6 +750,44 @@ open class SymbolResolver(ctx: TranslationContext) : ComponentPass(ctx) {
         }
     }
 
+    private fun handleOverloadedOperator(op: HasOverloadedOperation) {
+        val result = resolveOperator(op)
+        val decl = result?.bestViable?.singleOrNull() ?: return
+
+        // If the result was successful, we can replace the node
+        if (result.success == SUCCESSFUL && decl is OperatorDeclaration && op is Expression) {
+            val call = operatorCallFromDeclaration(decl, op)
+            walker.replace(op.astParent, op, call)
+        }
+    }
+
+    private fun resolveOperator(op: HasOverloadedOperation): CallResolutionResult? {
+        val language = op.language
+        val base = op.operatorBase
+        if (language !is HasOperatorOverloading || language.isPrimitive(base.type)) {
+            return null
```

Comment:
Are we sure that we cannot overload an operation on a primitive type in a language? Or am I missing something?